src/main/kotlin/com/openlattice/rhizome/service/ContinuousRepeatingTaskService.kt

```diff
@@ -0,0 +1,136 @@
+        logger.info("Starting enqueuer task for $className ")
+        executor.submit {
+            while (true) {
+                try {
+                    task.sourceSequence()
+                            .filter {
+                                logger.debug("Queueing candidate {}", it)
+                                filterEnqueued( it )
+                            }.forEach {
+                                candidateQueue.put(it)
+                            }
+                } catch (ex: Exception) {
+                    logger.info("Encountered error while enqueuing candidates for $className task.", ex)
+                }
+            }
+        }
+    } else {
+        logger.info("Skipping $className enqueue task as it is not enabled.")
+        null
+    }
+
+    @Suppress("UNUSED")
+    private val worker = if ( task.workerEnabledCheck() ) {
+        logger.info("Starting worker task for $className ")
+        executor.submit {
+            while ( true ) {
+                try {
+                    ChunkedQueueSequence(candidateQueue, workerChunkSize)
+                            .map { candidateChunk ->
+                                candidateChunk.filter {
+                                    filterInvalidStates(it)
+                                }.map { candidate ->
+                                    limiter.acquire()
+                                    executor.submit {
+                                        try {
+                                            logger.info("Operating on {}", candidate)
+                                            task.operate(candidate)
+                                        } catch (ex: Exception) {
+                                            logger.error("Unable to operate on $candidate. ", ex)
+                                        } finally {
+                                            finishedProcessing(candidate)
+                                            limiter.release()
+                                        }
+                                    }
+                                }.forEach { job ->
+                                    job.get(task.getTimeoutMillis(), TimeUnit.MILLISECONDS)
+                                }
+                            }
+                } catch ( ex: Exception) {
+                    logger.info("Encountered error while operating on candidates.", ex )
+                }
+            }
+        }
+    } else {
+        logger.info("Skipping $className worker task as it is not enabled.")
+        null
+    }
+
+    /**
+     * @return Boolean indicating whether candidate was successfully queued
+     */
+    private fun filterEnqueued( candidate: T ): Boolean {
+        return statusMap.putIfAbsent(candidate, QueueState.QUEUED) != null
+    }
+
+    /**
+     * Return true for properly queued candidates
+     */
+    private fun filterInvalidStates( candidate: T ): Boolean {
+        val currStatus = statusMap.get( candidate )
```

Comment:
is there a race condition because this isn't implemented as entry processor? i.e the same object may end up getting processed twice in two threads.